### DISCUSSION TITLE
Possible to get TermSet by name?

### DISCUSSION THREAD

**Discussion by aaronmorgan** [pnp/pnpcore#918] [Q&A] [answered]
Hi, I'm iterating over the TermSet collection to find the one I need then processing it's Term objects. 

I haven't found a way to get a TermSet by name and so am doing the following. While it works it feels suboptimal and like I'm missing something obvious. 

1. Is this the correctly way to go about getting a TermSetCollection by name?
2. The documentation frequently refers to using ids when getting TermSet, Terms etc and that makes me wonder if I haven't implemented this correctly and should be coming at it from the perspective of using ids. Only, I don't know the ids until I query the TermStore so...?

```
var termStore = await _pnpContext.TermStore.GetAsync();
var termGroup = await termStore.Groups.GetByNameAsync(termGroupKey);

await termGroup.LoadAsync(p => p.Sets);

foreach (var termSet in termGroup.Sets.AsRequested())
{
    var localizedName = termSet.LocalizedNames.FirstOrDefault(x => x.Name == termSetKey);
    if (localizedName == null) { continue; }

    await termSet.LoadAsync(p => p.Terms);
    await LoadTermsRecursive(termSet.Terms);

    break;
}
```

Thank you.

**Comment by jansenbe** [comment, marked as answer]
@aaronmorgan : currently there are no Graph APIs that enable you to get a termset by name without knowing the group it's in. Only for ID based retrieval this currently is possible. If your "entry point" is a list item/field then you should be able to get the termset and term id from there and use that to load the needed terms.

The feature team is aware of this and changes are being evaluated, but nothing will be available in the short term
